apps/gitness/src/pages/project-settings/components/FormDialogProjectDelete.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from 'react'
+import {
+  Button,
+  Input,
+  Icon,
+  AlertDialog,
+  AlertDialogTrigger,
+  AlertDialogContent,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogCancel
+} from '@harnessio/canary'
+import { FormFieldSet, MessageTheme } from '@harnessio/playground'
+import { useForm } from 'react-hook-form'
+import { z } from 'zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+
+const projectDeleteSchema = z.object({
+  verification: z.string().min(1, { message: 'Please type the DELETE to verify' })
+})
+
+type ProjectDeleteFields = z.infer<typeof projectDeleteSchema>
+
+interface FormProjDeleteProps {
+  handleDeleteProject: () => void
+  isDeleteSuccess: boolean
+  isDeleting: boolean
+}
+//delete project form with dialog
+export const FormDialogProjectDelete = ({ handleDeleteProject, isDeleteSuccess, isDeleting }: FormProjDeleteProps) => {
+  const {
+    register,
+    formState: { errors },
+    watch
+  } = useForm<ProjectDeleteFields>({
+    resolver: zodResolver(projectDeleteSchema)
+  })
+
+  const [isDialogOpen, setIsDialogOpen] = useState(false)
+
+  const verificationValue = watch('verification')
+
+  const typeCheck = (value: string) => {
+    return value === 'DELETE'
+  }
+
+  const handleDelete = () => {
+    handleDeleteProject()
+    if (isDeleteSuccess) {
+      setTimeout(() => {
+        setIsDialogOpen(false) // Close the dialog
+      }, 2000) // Redirect after 2 seconds
+    }
```

Comment:
Unsure if we need to wait to close dialog?

Reply:
again from the playground status just migrate from there

Reply:
Will double check if we still follow the flow in the playground but I assume it's a guidance there